mono/utils/mono-threads-windows.c

```diff
@@ -412,6 +412,14 @@ mono_native_thread_os_id_get (void)
 	return (guint64)GetCurrentThreadId ();
 }
 
+gint32
+mono_native_thread_processor_id_get (void)
+{
+	PROCESSOR_NUMBER proc_num;
+	GetCurrentProcessorNumberEx (&proc_num);
+	return ((proc_num.Group << 6) | proc_num.Number);
```

Comment:
Depending, it might be preferable to return `GetCurrentProcessorNumber`.
1. Are processor groups 0 .. n-1?
2. Are processor numbers 0 .. -1?
I guess callers could remove do % 64 or something.
I guess from the other comments, this matches CoreCLR.

Reply:
Yep, I just took their logic. The idea behind this API as I understand it is to enable data structures with greater cache locality, so I think you want to preserve the group in some fashion. Not sure if it's guaranteed to be [0..n) - I think it is on bare metal but if you're in a hypervisor all bets may be off?